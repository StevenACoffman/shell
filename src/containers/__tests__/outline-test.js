import React from "react";
import OutlineComponent from "../outline_container.jsx";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

const mockStore = configureStore([]);
const store = mockStore({
    thesis: {thesis_value:""},
    sections: [{
        id:0,
        name: "",
        notes: "",
        citations: [],
        canMoveSectionUp: false,
        canMoveSectionDown: false,
        selectedListItems: []
    }],
    list: {
        citationStyle: "",
        listItems: []
    }
});

it("renders correctly", () => {
    const tree = renderer.create(
    <Provider store={store}>
      <OutlineComponent/>
    </Provider>
  ).toJSON();
    expect(tree).toMatchSnapshot();
});
