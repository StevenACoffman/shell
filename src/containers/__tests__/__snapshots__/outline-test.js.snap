exports[`test renders correctly 1`] = `
<div>
  <div
    className="row">
    <div>
      <span>
        <label
          htmlFor="citation-format-picker">
          Citation Format:
        </label>
        <select
          id="citation-format-picker"
          onChange={[Function]}
          value="">
          <option
            value="mla">
            MLA
          </option>
          <option
            value="apa">
            APA
          </option>
          <option
            value="chicago">
            CHICAGO
          </option>
        </select>
      </span>
      <ul
        className="list-jstor hide" />
    </div>
  </div>
  <div
    className="row">
    <div
      className="small-12 columns">
      <div
        id="thesis-component">
        <label
          htmlFor="thesis-body">
          Thesis Statement
        </label>
        <textarea
          id="thesis-body"
          onChange={[Function]}
          placeholder="Placeholder..."
          rows="5"
          type="text"
          value="" />
      </div>
      <div>
        <div
          className="row">
          <div
            className="small-9 columns">
            <div>
              <label
                htmlFor="name_0">
                Section Name
              </label>
              <input
                id="name_0"
                onChange={[Function]}
                type="text"
                value="" />
            </div>
            <div>
              <label
                htmlFor="notes_0">
                Section Notes
              </label>
              <textarea
                id="notes_0"
                name="notes"
                onChange={[Function]}
                rows="5"
                value="" />
            </div>
            <div>
              <div>
                <span
                  className="subheader">
                  Citations
                </span>
                <ul
                  className="list-jstor"
                  id="citations_0" />
              </div>
            </div>
            <div>
              <button
                className="btn-link"
                data-reveal-id="sectionModal_0"
                id="add_citations_section_0">
                + Add Citation From List
              </button>
              <div
                className="row">
                <div
                  className="small-12">
                  <section
                    className="pl-section">
                    <div
                      className="reveal-modal"
                      data-reveal={true}
                      id="sectionModal_0">
                      <header>
                        <h2>
                          Add Citations From List
                        </h2>
                      </header>
                      <div
                        className="reveal-body">
                        <div>
                          <h2>
                            Select Citations To Add
                          </h2>
                          <hr />
                          <input
                            className="mylists-single-option"
                            type="checkbox" />
                          Title
                          <hr />
                          <ul />
                          <div>
                            <button
                              className="button button-jstor"
                              onClick={[Function]}>
                              Add to Section
                            </button>
                            <button
                              className="btn-link"
                              onClick={[Function]}>
                              Cancel
                            </button>
                          </div>
                        </div>
                        <a
                          aria-label="Close Overlay"
                          className="close-reveal-modal"
                          href=""
                          onClick={[Function]}>
                          Ã— 
                          <span
                            className="visuallyhidden">
                            Close Overlay
                          </span>
                        </a>
                      </div>
                    </div>
                  </section>
                </div>
              </div>
            </div>
          </div>
          <div
            className="small-3 columns">
            <div
              className="mtxl action-buttons">
              <div>
                <button
                  className="button button-jstor secondary disabled"
                  disabled={true}
                  id="move_section_up_button_0"
                  onClick={[Function]}
                  type="button">
                  Move Section Up
                </button>
              </div>
              <div>
                <button
                  className="button button-jstor secondary disabled"
                  disabled={true}
                  id="move_section_down_button_0"
                  onClick={[Function]}
                  type="button">
                  Move Section Down
                </button>
              </div>
              <button
                className="button button-jstor"
                id="delete_section_0"
                onClick={[Function]}>
                Delete Section
              </button>
            </div>
          </div>
        </div>
      </div>
      <button
        className="button button-jstor"
        id="add-section"
        onClick={[Function]}
        type="button">
        + Add Section
      </button>
    </div>
  </div>
</div>
`;
